src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -90,36 +73,60 @@ public void removeMetric(KafkaMetric metric) {
         metrics.remove(metric.metricName());
     }
 
-    String metricName(MetricName metricName) {
-        String prefix = this.prefix
-                .replace('.', '_')
-                .replace('-', '_')
-                .toLowerCase(Locale.ROOT);
-        String group = metricName.group()
-                .replace('.', '_')
-                .replace('-', '_')
-                .toLowerCase(Locale.ROOT);
-        String name = metricName.name()
-                .replace('.', '_')
-                .replace('-', '_')
+    /**
+     * Called when the Prometheus server scrapes metrics.
+     * @return metrics that match the configured allow list
+     */
+    @Override
+    public MetricSnapshots collect() {
+        Map<String, GaugeSnapshot.Builder> gaugeBuilders = new HashMap<>();
```

Comment:
Unfortunately the `MetricSnapshotBuilder` interface does not have a `build()` method. I opened a [PR](https://github.com/prometheus/client_java/pull/969) against [prometheus/client_java](https://github.com/prometheus/client_java) to change that.

In the meantime, I think we have to keep a collection per metric type.